src/topology/sheaves/Godement.lean

```diff
@@ -0,0 +1,77 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import topology.sheaves.sheaf
+import topology.sheaves.limits
+import topology.sheaves.skyscraper
+import topology.sheaves.stalks
+import category_theory.preadditive.injective
+
+/-!
+# Godement resolution
+
+For a presheaf `𝓕 : (opens X)ᵒᵖ ⥤ C`, we can embedded `𝓕` into a sheaf `∏ₓ skyscraper(𝓕ₓ)` where
+`x` ranges over `X` and `𝓕 ⟶ ∏ₓ skyscraper(𝓕ₓ)` is mono.
+
+## Main definition
+* `godement_presheaf`: for a presheaf `𝓕`, its Godement presheaf is `∏ₓ skyscraper(𝓕ₓ)`
+* `to_godement_presheaf`: the canonical map `𝓕 ⟶ godement_presheaf 𝓕` sending `s : 𝓕(U)` to a
+  bundle of stalks `x ↦ sₓ`.
+-/
+
+noncomputable theory
+
+section presheaf
+
+open Top
+open topological_space
+open category_theory
+open category_theory.limits
+
+universes u v
+
+variables {X : Top.{u}} {C : Type u} [category.{u} C]
+variables [has_limits C] [has_terminal C] [has_colimits C]
```

Comment:
```suggestion
variables [has_limits C] [has_colimits C]
```
`has_limits` implies`has_terminal`.

Reply:
Actually, `has_products` suffice for this file, right?

Reply:
we still need `has_colimits C` for `stalks`

Reply:
You should only need `has_filtered_colimits` for that, right?

Reply:
I just caught a bad flu and I will get back to this as soon as possible. Sorry for the delay.

Reply:
`stalk_functor` assumes `has_colimits`, so `has_colimits` cannot be relaxed

Reply:
> Actually, `has_products` suffice for this file, right? But we lack an instance from `has_products` to `has_terminal`. Maybe `has_products_of_shape X` + `has_terminal`?
> 
> We have [category_theory.limits.functor_category_has_limits_of_shape](https://leanprover-community.github.io/mathlib_docs/category_theory/limits/functor_category.html#category_theory.limits.functor_category_has_limits_of_shape) but since `presheaf` is not a reducible def we need to manually add one more instance to topology/sheaves/limits for Lean to infer that the presheaf category also has limits of shape `X`.

`limit_is_sheaf` requires `has_limits C`